ng-dev/utils/testing/bazel-env.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {join} from 'path';
+
+/**
+ * Temporary directory which will be used as project directory in tests. Note that this environment
+ * variable is automatically set by Bazel for tests. Bazel expects tests "not attempt to remove,
+ * chmod, or otherwise alter [TEST_TMPDIR]," so a subdirectory path is used to be created/destroyed.
+ */
+export const testTmpDir: string = join(process.env['TEST_TMPDIR']!, 'dev-infra');
```

Comment:
Unfortunately this is prone to errors as the `dev-infra` folder inside the Bazel test tmpdir does not necessarily exist (resulting in the issues I had when I ran tests on my fresh Bazel output base).

We should expose a method for ensuring it exists, but I foresee cases where we accidentally use `testTmpDir` without ensuring it exists. Maybe it should be only possible to retrieve the test tmp dir through a function like `prepareTestTmpDir()` where it returns the path then.